### PR TITLE
NW6 | Areeb Sattar | JS3 Module | [TECH ED] TV Show Project - Part 3/4 | Week 3

### PR DESCRIPTION
**Pull request by areebsattar**
## Learners, PR Template

Self checklist

- [ ] I have committed my files one by one, on purpose, and for a reason
- [ ] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [ ] I have tested my changes
- [ ] My changes follow the [style guide](https://syllabus.codeyourfuture.io/guides/code-style-guide/)
- [ ] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -85,15 +117,49 @@ function createEpisodeListItem(episode) {
   return episodeListItem;
 }
 
+function fillShowList() {
```
Comment (rafsanjani) [inline script.js]: A more conventional name for this function could be `populateShowSelector`

Comment (areebsattar) [reply script.js]: done

File: script.js [hunk 2]
```diff
@@ -1,38 +1,69 @@
...
+      console.log(data);
+      return data;
+    })
+    .catch((error) => {
+      console.error(error);
+      throw error;
+    });
 }
 
 setTimeout(function timeout() {
```
Comment (AltomHussain) [inline script.js]: You should not be using set timeout to fetch data you should just use async and a wait that will do the job for you!

File: script.js [hunk 3]
```diff
@@ -63,13 +94,14 @@ function handleSearchInput(event) {
...
-  console.log(state.selectedEpisodeId);
-  renderBySelect();
+function createShowListItem(show) {
+  const showListItem = document
+    .getElementById("show-list")
+    .content.cloneNode(true);
+  const option = showListItem.querySelector("option");
+  option.textContent = show.name;
+  option.setAttribute("value", show.id);
+  return showListItem;
```
Comment (AltomHussain) [inline script.js:91]: Very nice work!

File: script.js [hunk 4]
```diff
@@ -85,15 +117,49 @@ function createEpisodeListItem(episode) {
...
+
+function populateEpisodeSelector() {
+    episodeSelector.textContent = "";
+    const defaultOption = document.createElement("option");
+    defaultOption.selected = true;
+    defaultOption.disabled = true;
+    defaultOption.innerHTML = "Select an option";
+    document.getElementById("episode-selector").append(defaultOption);
   const episodes = state.episodes;
   for (e of episodes) {
```
Comment (AltomHussain) [inline script.js]: Can you rename `e` variable to more descriptive name?

File: script.js [hunk 5]
```diff
@@ -85,15 +117,49 @@ function createEpisodeListItem(episode) {
...
+  state.selectedShowId = event.target.value;
+  fetchEpisodes();
+}
+
+const episodeSelector = document.getElementById("episode-selector");
+episodeSelector.addEventListener("change", handleEpisodeSelection);
+
+function handleEpisodeSelection(event) {
+  state.selectedEpisodeId = event.target.value;
+  console.log(state.selectedEpisodeId);
```
Comment (AltomHussain) [inline script.js]: No need for the console.log to be pushed.

File: script.js [hunk 6]
```diff
@@ -122,10 +186,20 @@ function renderByFilter(filterFunction) {
   const episodeCards = filteredEpisodes.map(createEpisodeCard);
   rootElem.append(...episodeCards);
 
+  console.log(`${state.episodes} => before filter`);
+  console.log(`${filteredEpisodes} => After filter`);
```
Comment (AltomHussain) [inline script.js]: Again console.logs

**Review by AltomHussain — COMMENTED** [review]
Good work, but one of the issues I found looking at your repo link is that once I have selected an episode I cannot go back to all episodes via the dropdown. You should have the first option in the dropdown as "Select All Episode" so you can select that and it should take you back to the all episode. 
image

I know you have the "Show All episodes " in the page but can you add what I mentioned in the dropdown because a user will expect you to have this functionality as part of dropdown 

image



Also,  the responsiveness is not really great. So in mobile view your page does not show the data properly.  The header should be adjusted properly according to the chosen view. Also,  when a user displays your page in a mobile view it should only show one car on the view board to make it easy to see.
image

File: script.js (hunk 2 above)
Comment (areebsattar) [reply script.js]: I will update this, it was added by Rabia but I understand your explanation. Thank you

File: script.js (hunk 4 above)
Comment (areebsattar) [reply script.js]: Oops, looks like I updated them in my Repo but forgot to do that here, but I have made the changes now

File: script.js (hunk 2 above)
Comment (RbAvci) [reply script.js]: Ive put this for testing, slow response from the API.

**Comment by RbAvci** [comment]
Thanks for your work, Areeb!

**Pull request merged**